docs/gates.md

```diff
@@ -1,58 +1,235 @@
-## Gates
+# Gates
 
 A ``Gate`` is an operation that can be applied to a collection of
 qubits (objects with a ``Qid``).  ``Gates`` can be applied
 to qubits by calling their ``on`` method, or, alternatively
 calling the gate on the qubits.  The object created by such calls
 is an ``Operation``.
```

Comment:
Starting the paragraph with "A Gate is an operation that..." and ending with the definition of `Operation` is awkward and circular. 
If `Operation` is the more fundamental concept, let's start with defining that and then defining `Gate` as a operation factory that, given input qubits, generates its associated operation.
If we want to keep `Gate` as fundamental and `Operation` as derived from Gate+qubits, change the first line to not contain the word "operation".

Reply:
Changed the first line to use 'effect' instead of operation.  I also added the explanation of a gate as an operation factory to the end of the paragraph.  I figured that it is best to explain the concepts in several ways.